### DISCUSSION TITLE
CLI: Output forwards to Stderr rather than Stdout

### DISCUSSION THREAD

**Discussion by InterestingBrainPoops** [BattlesnakeOfficial/feedback#241] [Bug Reports]
Freethrill recently noticed that the CLI outputs all of its info to stderr, and since theres nothing going to stdout, it seems that the expected (and preferred) behaviour is to output to stdout rather than stderr.

**Comment by bvanvugt** [comment]
Ah, yes. Interesting. I think this was a design decision at some point, but I do not recall the details. 

I'd be in favor of changing it, but wouldn't want to do that without some exploration first. Maybe @robbles has some insight?

**Reply by rob-odwyer** [reply]
I think the intention here was to output logging to stderr, and machine parseable output to stdout. I'm not sure if anything actually made it into the second category though.

**Reply by rob-odwyer** [reply]
I think we generally would not want to just switch everything over to stdout. As I understand it, the main reason to do that would be to make it easier to pipe the output to another command, which implies you're relying on the format in some way. In that case, adding some new log lines might break things. If we select a few things that make sense to be going to stdout and without any logging prefixes, etc. I'm on board with that.
